lib/db/mongodb/errors.go

```diff
@@ -9,15 +9,16 @@ const (
 )
 
 func HasErrorCode(err error, code int) bool {
-	we, ok := err.(mongo.WriteException)
-	if !ok {
-		return false
-	}
-
-	for _, err := range we.WriteErrors {
-		if err.Code == code {
-			return true
+	switch v := err.(type) {
+	case mongo.WriteException:
```

Comment:
Golang doesn't fall through by default, you need to add the `fallthrough` key word :P  But I don't think that works on type switches :S 

Hopefully I am wrong tho

Reply:
It does!